src/offers/lnd_requests.rs

```diff
@@ -364,6 +366,34 @@ pub async fn send_invoice_request(
     Ok((contents, send_instructions))
 }
 
+pub(crate) async fn connect_to_peer_with_retry(
```

Comment:
we should add a maximum of retries in this one, we don't want it to run it for ever